### PR TITLE
Update the README and CONTRIBUTING guide to prepare for first release

### LINKED ISSUE #159: OSS suite has a versioning explanation and policy
After significant work and debate, we arrived at a versioning policy for the Conjur OSS Suite that is directly dependent upon the the Conjur version included in the release.  Essentially, it will be `x.y.z+suite.n`, where `x.y.z` is the Conjur version and `n` is an integer incremented for each suite release using the same Conjur version.  The latter component will likely not be incremented often.  This is a fairly straightforward and easy to understand scheme, but it still needs to be communicated to both user and contributors of this repo.  With that in mind, I would suggest addition of the following:

1. [x] Add a small section to the `README` explaining what our versions mean and how they relate to Conjur.  This is only intended for users.
2. [x] Add larger section to the `CONTRIBUTING` file explaining how a version number is determined, so that contributors conducting releases can decide which version to use for each release.
3. [x] File a separate issue so that the content in first todo/bullet above is also added to the docs.conjur.org site.  This will ensure that users understand our versioning, regardless of how they consume the suite.
4. [ ] (Optional) As said above, significant thought was put into this scheme.  I think there is value in publishing this in a public place.  Maybe we need a `designs` directory or it could go into the wiki.

### LINKED ISSUE #160: README briefly explains purpose/value of the project
At the moment, the `README` for this repo is too terse, as it's aimed primarily at contributors working to complete the first release.  It has served this process well, but as we prepare for the first release, it needs to provide a bit more context for users of the suite.  Thus, I would propose the following changes:

* [x] Add a section at the top of the doc that explains what this repo is and why someone might care about it.  Likely, this content can be largely lifted from the [design doc](https://github.com/cyberark/conjur/blob/master/design/oss_suite_release.md).
* [x] Add a section that explains to users how that might use this repo.  This might largely be a link to the "Releases" page.
* [x] Rename or reposition the current usage section that it's clear why someone would execute the changelog CLI.
* [ ] (Optional) Edit the [cyberark/conjur](https://github.com/cyberark/conjur) `README` file so that it has a link and brief description of the suite.

### PR DESCRIPTION
**Pull request by izgeri**
Resolves #160 
Resolves #159 

Updates the README with info about the purpose of the project and its versioning scheme, and adds utility usage info and detailed versioning instructions to CONTRIBUTING.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -75,6 +109,55 @@ $ go test -v -short ./...
...
+- If this is the first suite release for Conjur version `1.x.y`, then the suite
+  release will be versioned `1.x.y+suite.1`, where `1.x.y` matches the included
+  Conjur core version.
+
+- At the next point in time when there is sufficient content in the Conjur OSS
+  suite that stakeholders determine a suite release is merited:
+  - If the Conjur core version in the suite is still `1.x.y`, then the next suite
+    release version will be `1.x.y+suite.2`.
+  - If the Conjur core version has changed to `1.n.m`, then the next suite
+    release version will be `1.n.m+suite.1`.
```
Comment (alexkalish) [inline CONTRIBUTING.md:134]:
```suggestion
- Subsequent suite releases using the same Conjur OSS version require that the suite version build component be incremented.  For example, a second suite release using Conjur `1.2.3` should be versioned `1.2.3+suite.2`, as the first one would have been versioned as `1.2.3+suite.1`.
```
Does read better?  I feel like it covers the cases, but correct me if I'm wrong.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -75,6 +109,55 @@ $ go test -v -short ./...
...
+
+Additional notes:
+- If Conjur changes its version with a new **minor or patch** release, we _may_
+  have a new suite release, but it is not required unless the stakeholders agree.
+- If Conjur changes its version with a new **major** release, we _must_ have a
+  new suite release, and stakeholders will determine how long we will continue
+  to support old version of Conjur in the suite.
+- In any Conjur suite release, if the Conjur core version is the same as the last
+  suite release, the `.z` digit in the suite version will be incremented.
```
Comment (alexkalish) [inline CONTRIBUTING.md:143]: Isn't this restating what you have above?  Do you think this extra clarification is needed?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -75,6 +109,55 @@ $ go test -v -short ./...
...
+  it in the next release that follows the deprecation announcement.
+- If a new component is added to the suite, stakeholders will determine
+  when to include the new component in the suite release.
+- Changes to the Conjur server version _may_ result in changes to the suite version
+  in a subsequent release, but changes to the suite version _never_ result in
+  changes to the Conjur server version.
+
+In each line above, when we refer to "stakeholders" we are referring to the
+maintainers of the components in the suite and CyberArk product management.
+Anyone can request a new suite release, if they believe it is merited.
```
Comment (alexkalish) [inline CONTRIBUTING.md:157]: Love this line!

File: README.md [hunk 4]
```diff
@@ -1,45 +1,71 @@
 # conjur-oss-suite-release
-Latest stable releases of the Conjur OSS suite
+
+Building the official releases of the [Conjur OSS suite](https://cyberark.github.io/conjur)
 
 **THIS REPO IS UNDER CONSTRUCTION, AND THERE ARE NO SUITE RELEASES AVAILABLE YET**
```
Comment (alexkalish) [inline README.md:5]: Can we remove this line now?

File: README.md [hunk 5]
```diff
@@ -1,45 +1,71 @@
...
 
 _For more info on the planned work, please see our [design doc](https://github.com/cyberark/conjur/blob/master/design/oss_suite_release.md)
 and monitor the [github issues](https://github.com/cyberark/conjur-oss-suite-release/issues)._
 
+Once we have a first release with release notes, we'll be adding a link to the
```
Comment (alexkalish) [inline README.md:10]: I guess we have a chicken and the egg issue here, as we don't have a public URL yet?

File: README.md [hunk 6]
```diff
@@ -1,45 +1,71 @@
...
+- Learn about new integrations or tools that work with Conjur
+- Find out which version of Conjur to use, and which corresponding versions of
+  the suite components are compatible with it
+
+If you're using Conjur OSS, we **strongly recommend** that you reference the latest
+suite release (link TBA) to determine which version of Conjur to use, and which corresponding
+versions of the suite are compatible with it.
+
+### A note on versioning
+The core of each Conjur OSS Suite release is the [Conjur server](https://github.com/cyberark/conjur).
```
Comment (alexkalish) [inline README.md:47]: Do we use the term `server` elsewhere for Conjur?  Feels like slightly odd word.

File: README.md [hunk 7]
```diff
@@ -1,45 +1,71 @@
...
+In light of this, we decided to tie our versioning of the suite to the _Conjur
+server version_.
+
+This means that if Conjur is included in the suite at version `1.x.y`, then the
+suite release will be versioned as `1.x.y+suite.z` (following the [semver standard](https://semver.org/)
```
Comment (alexkalish) [inline README.md:52]: Can you replace the `1` with `w` throughout?

```suggestion
suite release will be versioned as `w.x.y+suite.z` (following the [semver standard](https://semver.org/)
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -75,6 +109,55 @@ $ go test -v -short ./...
 
 ## Releasing
 
+### Notes on versioning
+
+We version the suite using the syntax `1.x.y+suite.z`, where `1.x.y` is the version of
```
[the remark this answers is not in the document]
Comment (izgeri) [reply CONTRIBUTING.md:114]: I considered that, but I think it's actually more readable in this format and I don't think (as it's an example) we need to update it for v2

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -75,6 +109,55 @@ $ go test -v -short ./...
...
+the Conjur server included in the suite.
+
+This works as follows:
+
+- When there is sufficient content in the Conjur OSS suite that stakeholders
+  determine a suite release is merited, a new suite release is prepared with version
+  `1.x.y+suite.z`, where `1.x.y` is the version of [Conjur](https://github.com/cyberark/conjur)
+  included in the release.
+  - Note: suite releases will not necessarily happen for every Conjur core release.
```
[the remark this answers is not in the document]
Comment (izgeri) [reply CONTRIBUTING.md:123]: I actually think Conjur OSS sort of sounds like the suite now, so I'm not sure. We call it "Conjur Core" here: https://cyberark.github.io/conjur/

File: CONTRIBUTING.md (hunk 2 above)
Comment (izgeri) [reply CONTRIBUTING.md:143]: it might not be, but it doesn't hurt?

File: README.md (hunk 5 above)
Comment (izgeri) [reply README.md:10]: yup

File: README.md (hunk 4 above)
Comment (izgeri) [reply README.md:5]: I'd like to wait until we have a published release

File: README.md (hunk 6 above)
Comment (izgeri) [reply README.md:47]: yes, again on cyberark.github.io/conjur. `core` and `server` are both terms PM has used to describe the oss architecture; `core` is the label and `server` is just descriptive, as it is a literal server

File: README.md (hunk 7 above)
Comment (izgeri) [reply README.md:52]: I would prefer not

File: CONTRIBUTING.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (alexkalish) [reply CONTRIBUTING.md:114]: Fine to leave for now, though I did ping @sgnn7 for his opinion in chat.

File: CONTRIBUTING.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (alexkalish) [reply CONTRIBUTING.md:123]: Ooo... yeah, this is a good point and speaks to some naming confusion.  Lets leave it for now, but would love to iron this naming out and update in all the places.

**Pull request merged**